readthedocs/proxito/middleware.py

```diff
@@ -367,11 +369,33 @@ def _get_https_redirect(self, request):
 
         return None
 
+    def add_resolver_headers(self, request, response):
+        if hasattr(request, "unresolved_url") and request.unresolved_url is not None:
+            # TODO: add more ``X-RTD-Resolver-*`` headers
+            header_value = escape(request.unresolved_url.filename)
+            try:
+                # Use Django internals to validate the header's value before injecting it.
+                ResponseHeaders({})._convert_to_charset(
+                    header_value,
+                    "latin-1",
+                    mime_encode=True,
+                )
```

Comment:
Why is this necessary? Do we have examples of files that aren't valid for header values?

Reply:
This is required to avoid Django complaining about a HTTP header value not supported. I found this because of tests with weird filenames on their URL: `http://project.dev.readthedocs.io/http:/%0D/my.host/path.html`. I took a look at the Django's internals to understand how they perform the validation of the values and copied it here.

Unfortunately, we can't just try/catch the line where we do the assigning since it doesn't fail _there_ --that's why I'm performing this validation _before_ the assignment.